src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -36,11 +36,25 @@ internal static SwitchBinder Create(Binder next, SwitchStatementSyntax switchSyn
                 // lead to a translation that fully supports edit-and-continue, so it delegates to the C# 6
                 // binder when it can. The "testV7SwitchBinder" feature flag forces the use of the C# 7 switch binder
                 // for all operations; we use it to enhance test coverage.
-                (parseOptions?.IsFeatureEnabled(MessageID.IDS_FeaturePatternMatching) != false || parseOptions?.Features.ContainsKey("testV7SwitchBinder") != false)
+                (parseOptions?.IsFeatureEnabled(MessageID.IDS_FeaturePatternMatching) != false ||
+                 parseOptions?.Features.ContainsKey("testV7SwitchBinder") != false ||
+                 switchSyntax.HasErrors && HasPatternSwitchSyntax(switchSyntax))
```

Comment:
This is the key to the fix: use a V7 switch binder if there are any patterns in the switch, even in language version 6.